include/metall/detail/file.hpp

```diff
@@ -27,56 +27,29 @@
 #include <vector>
 #include <thread>
 #include <atomic>
-#include <functional>
-
-#ifdef __has_include
-
-// Check if the Filesystem library is available or disabled by the user
-#if __has_include(<filesystem>) && !defined(METALL_DISABLE_CXX17_FILESYSTEM_LIB)
 #include <filesystem>
-#else
-#ifdef METALL_VERBOSE_SYSTEM_SUPPORT_WARNING
-#warning "The Filesystem library is not available or disabled by the user."
-#endif
-#endif
-
-#else  // __has_include is not defined
-
-#ifdef METALL_VERBOSE_SYSTEM_SUPPORT_WARNING
-#warning \
-    "__has_include is not defined, consequently disable the Filesystem library."
-#endif  // METALL_VERBOSE_SYSTEM_SUPPORT_WARNING
-
-#endif  // #ifdef __has_include
 
 #include <metall/logger.hpp>
 
 namespace metall::mtlldetail {
 
-#if defined(__cpp_lib_filesystem) && \
-    !defined(METALL_DISABLE_CXX17_FILESYSTEM_LIB)
-namespace {
-namespace fs = std::filesystem;
-}
-#endif
-
 inline bool os_close(const int fd) {
   if (::close(fd) == -1) {
-    METALL_ERRNO_ERROR("close");
+    METALL_ERRNO_WARN("close");
     return false;
   }
   return true;
 }
 
 inline bool os_fsync(const int fd) {
   if (::fsync(fd) != 0) {
-    METALL_ERRNO_ERROR("fsync");
+    METALL_ERRNO_WARN("fsync");
     return false;
   }
   return true;
 }
 
-inline bool fsync(const std::string &path) {
+inline bool fsync(const std::filesystem::path &path) {
```

Comment:
When would os_close in th fsync return false?

Reply:
if the file could not be closed. See #Errors section here: https://www.man7.org/linux/man-pages/man2/close.2.html

Reply:
In general all systemcalls can fail for various reasons

Reply:
And we don't track what the actual reason was. so, effectively we end up in an unrecoverable state, right?

Reply:
I would consider a file failing to close to be ignorable it only means that we use an additional file descriptor. Which is a leak, but usually unproblematic. Fsync failing is usually also not usually problematic, the data will get flushed by the kernel eventually.

Realistically, close and fsync will probably never fail if there isn't something fundamentally broken on the system.